docs_src/design/adr/security/tbd-authentication-e2ee.md

```diff
@@ -0,0 +1,414 @@
+Even though the original JWT passes through the proxy,
+the Kong authentication plugins do not expose
+token introspection endpoints that the backend service
+could use to check token validity independently.
+
+The consequence of having an API gateway that performs
+all microservice authentication is that communication
+between EdgeX microservices running behind the API gateway
+are not authenticated in any way.
+EdgeX microservices are unable to distinguish
+malicious traffic that has evaded the API gateway
+from legitimate microservice traffic.
+
+## Proposed Design
+
+This ADR proposes an implementation of the
+[Microservice Authentication UCR](../../ucr/Microservice-Authentication.md)
+that uses an end-to-end authentication mechanism based on zero-trust networking.
+
+In this authentication model,
+EdgeX microservices connect directly to the zero-trust network overlay
+and communicate with other EdgeX microservices in a manner
+that is fully encrypted end-to-end and
+governed by a centrally managed network policy.
+In this model, an EdgeX microservice API never "listens"
+on a host-accessible network port.
+Instead, EdgeX microservices make outgoing connections
+to an OpenZiti router.
+Unlike the token-based authentication ADR,
+this end-to-end encryption ADR secures network traffic by default,
+and makes to possible to have a fully-distributed EdgeX implementation
+that spans the edge to the cloud.
+In the token-based authentication ADR,
+developers had to decide which routes were authenticated and which were not:
+in the end-to-end encryption design,
+the entire REST API of each microservice is secured in its entirety.
+
+This ADR prescribes two methods of onboarding to the OpenZiti fabric:
+
+1. A local EdgeX microservice, already having a Vault token
+   and exchanged it for a claims-based identity (JWT),
+   supplies the JWT to the OpenZiti Router
+   to establish establish a secure session
+   and establish a data plane connection to the zero-trust network.
+
+2. A remote EdgeX microservice or remote administrator,
+   having cached an OpenZiti identity (certificate and private key)
+   through some sideband mechanism,
+   connects directly to the OpenZiti router
+   and becomes a member of the zero-trust network.
+
+Under this ADR, EdgeX microservices will directly integrate with the OpenZiti SDK.
+Integration with the OpenZiti SDK enables EdgeX microservices
+to natively communicate on the OpenZiti zero-trust network.
+However, to be fully functional,
+and adopter must also run an OpenZiti controller,
+one or more OpenZiti router components,
+and zero or more OpenZiti tunnelers to support legacy applications.
+
+In a zero-trust architecture, an API gateway is no longer required
+for remote access to EdgeX microservice API's.
+If an API gateway is still desired,
+it is still possible to run the EdgeX 3.0 API gateway
+in conjunction with an OpenZiti tunneler to onboard
+API gateway traffic onto the zero-trust network on the backend.
```

Comment:
Does it make sense to mention the addtional 'reach' the overlay can now provide to cross cloud/cross site/remote maintainers? If edgex users are exclusively local, it might not be interesting to them but the ability to securely access their installation from 'anywhere' (or monitor it, etc) is a fantastic addition feature imo?

Reply:
Yes, I think so.  Being able to have reach across multiple EdgeX deployments seems a great manageability play.

Reply:
This has been addressed in the latest update.